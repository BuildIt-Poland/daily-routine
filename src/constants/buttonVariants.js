export const BRAG = 'brag';
export const CONFESS = 'confess';
export const FRONT_END = 'frontend';
export const BACK_END = 'backend';
export const DEV_OPS = 'devops';
