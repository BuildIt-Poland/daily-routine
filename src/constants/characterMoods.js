export const CONFUSED = 'confused';
export const CONFIDENT = 'confident';
export const SAD = 'sad';
