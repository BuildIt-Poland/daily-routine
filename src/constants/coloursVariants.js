export const FRONT_END_VARIANT = 'frontend';
export const BACK_END_VARIANT = 'backend';
export const DEV_OPS_VARIANT = 'devops';
export const CONFESS_VARIANT = 'confess';
export const BRAG_VARIANT = 'brag';
