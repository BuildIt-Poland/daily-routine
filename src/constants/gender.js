export const MALE_GENDER = 'male';
export const FEMALE_GENDER = 'female';
