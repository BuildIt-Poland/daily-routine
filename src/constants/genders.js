export const GENDER = 'gender';
export const MALE = 'male';
export const FEMALE = 'female';
