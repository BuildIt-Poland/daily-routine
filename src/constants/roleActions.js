export const DEFAULT = 'default';
export const BRAG = 'brag';
export const CONFESS = 'confess';
