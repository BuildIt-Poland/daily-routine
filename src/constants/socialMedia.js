export const FACEBOOK = 'facebook';
export const TWITTER = 'twitter';
