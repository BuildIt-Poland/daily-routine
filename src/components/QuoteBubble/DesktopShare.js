import React from 'react';

import { socialMediaVariant } from '../../types';
import { Facebook, Twitter } from '../Icons';
import Hitbox from '../Hitbox';
import { FACEBOOK, TWITTER } from '../../constants/socialMedia';

const socialMediaData = {
  [FACEBOOK]: {
    shareUrl: `http://www.facebook.com/sharer.php?u=${window.location.href}`,
    icon: Facebook,
    testID: 'facebook-share-button'
  },
  [TWITTER]: {
    shareUrl: `http://www.twitter.com/share?url=${window.location.href}`,
    icon: Twitter,
    testID: 'twitter-share-button'
  }
};

function DesktopShare({ variant }) {
  const { icon: Icon, shareUrl, testID } = socialMediaData[variant];

  const share = () =>
    window.open(
      shareUrl,
      'shareWindow',
      'toolbar=no,location=0,status=no,menubar=no,scrollbars=yes,resizable=yes,width=600,height=275'
    );

  return (
    <Hitbox onClick={share} data-testid={testID}>
      <Icon />
    </Hitbox>
  );
}

DesktopShare.propTypes = {
  variant: socialMediaVariant.isRequired
};

export default DesktopShare;
