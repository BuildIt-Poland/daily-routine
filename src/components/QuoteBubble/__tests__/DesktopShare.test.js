import React from 'react';
import { create } from 'react-test-renderer';
import { render, fireEvent, cleanup } from '@testing-library/react';

import DesktopShare from '../DesktopShare';
import { FACEBOOK, TWITTER } from '../../../constants/socialMedia';

afterEach(cleanup);

describe('COMPONENT - QuoteBubble DesktopShare', () => {
  it('renders correctly for Facebook variant', () => {
    const component = create(<DesktopShare variant={FACEBOOK} />);

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders correctly for Twitter variant', () => {
    const component = create(<DesktopShare variant={TWITTER} />);

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('calls Navigator.share API with proper title and url', () => {
    const { container } = render(<DesktopShare variant={FACEBOOK} />);

    fireEvent.click(container.querySelector('button'));

    const expectedShareUrl = `http://www.facebook.com/sharer.php?u=${window.location.href}`;
    const expectedWindowName = 'shareWindow';
    const expectedWindowFeatures =
      'toolbar=no,location=0,status=no,menubar=no,scrollbars=yes,resizable=yes,width=600,height=275';

    expect(window.open).toBeCalledWith(expectedShareUrl, expectedWindowName, expectedWindowFeatures);
  });
});
