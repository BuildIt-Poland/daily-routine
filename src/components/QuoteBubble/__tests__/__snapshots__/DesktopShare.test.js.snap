// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - QuoteBubble DesktopShare renders correctly for Facebook variant 1`] = `
.c0 {
  border: 0;
  margin: 0;
  padding: 0;
  background: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  min-width: 4rem;
  min-height: 4rem;
  cursor: pointer;
}

<button
  className="c0"
  onClick={[Function]}
>
  <svg
    aria-labelledby="facebook-share-title"
    height="20"
    viewBox="0 0 20 20"
    width="20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title
      id="facebook-share-title"
    >
      Share on Facebook
    </title>
    <path
      d="M20 10c0-5.523-4.477-10-10-10S0 4.477 0 10c0 4.991 3.657 9.128 8.438 9.878V12.89h-2.54V10h2.539V7.797c0-2.506 1.493-3.891 3.777-3.891 1.094 0 2.238.195 2.238.195v2.461h-1.261c-1.242 0-1.63.771-1.63 1.562V10h2.773l-.443 2.891h-2.33v6.988C16.343 19.128 20 14.991 20 10z"
      fill="#ccc"
    />
  </svg>
</button>
`;

exports[`COMPONENT - QuoteBubble DesktopShare renders correctly for Twitter variant 1`] = `
.c0 {
  border: 0;
  margin: 0;
  padding: 0;
  background: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  min-width: 4rem;
  min-height: 4rem;
  cursor: pointer;
}

<button
  className="c0"
  onClick={[Function]}
>
  <svg
    aria-labelledby="twitter-share-title"
    height="20"
    viewBox="0 0 20 20"
    width="20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title
      id="twitter-share-title"
    >
      Share on Twitter
    </title>
    <path
      d="M10 0C4.477 0 0 4.477 0 10s4.477 10 10 10 10-4.477 10-10S15.523 0 10 0zm5.025 8.105c.005.1.005.205.005.31 0 3.185-2.425 6.86-6.86 6.86-1.36 0-2.63-.4-3.685-1.09a4.847 4.847 0 0 0 3.57-.995 2.407 2.407 0 0 1-2.25-1.675 2.403 2.403 0 0 0 1.09-.04A2.414 2.414 0 0 1 4.96 9.11v-.03c.325.18.695.29 1.09.3a2.4 2.4 0 0 1-1.075-2.005c0-.44.12-.855.325-1.21a6.84 6.84 0 0 0 4.97 2.52 2.411 2.411 0 0 1 4.105-2.2c.55-.11 1.065-.31 1.53-.585a2.41 2.41 0 0 1-1.06 1.335c.49-.06.955-.19 1.385-.38a4.94 4.94 0 0 1-1.205 1.25z"
      fill="#ccc"
    />
  </svg>
</button>
`;
